rules/Php80/Rector/If_/NullsafeOperatorRector.php

```diff
@@ -246,6 +248,15 @@ private function shouldProcessAssignInCurrentNode(Assign $assign, Node $nextNode
             return false;
         }
 
+        if (
+            property_exists($assign->expr, self::VAR) &&
+            property_exists($nextNode->expr, self::VAR) &&
```

Comment:
This check is quite magical. PHPStan and Rector would not understand it and fail to refactor in the future.
It should be handled instanceof with specific classes.

Reply:
Well, I took inspiration from the code above. If it's a problem perhaps wider refactoring of the class would proof beneficial.

Reply:
You may can change the code something like this:

```php
    private function shouldProcessAssignInCurrentNode(Assign $assign, Node $nextNode): bool
    {
        if (! $nextNode instanceof Return_ && ! $nextNode instanceof Expression) {
            return false;
        }

        if ($nextNode->expr instanceof Assign) {
            return false;
        }

        if (! $assign->expr instanceof MethodCall) {
            return ! $this->valueResolver->isNull($nextNode->expr);
        }

        if (! $nextNode->expr instanceof MethodCall) {
            return ! $this->valueResolver->isNull($nextNode->expr);
        }

        if (! $this->nodeComparator->areNodesEqual($assign->expr->var, $nextNode->expr->var)) {
            return ! $this->valueResolver->isNull($nextNode->expr);
        }

        return false;
    }
```

after than, run the following commands:

```bash
# rectify
bin/rector process rules/Php80/Rector/If_/NullsafeOperatorRector.php

# fix cs 
composer fix-cs -- rules/Php80/Rector/If_/NullsafeOperatorRector.php

# rectify again to ensure all ok
bin/rector process rules/Php80/Rector/If_/NullsafeOperatorRector.php
```